nemo/collections/nlp/data/language_modeling/megatron/data_samplers.py

```diff
@@ -14,43 +14,84 @@
 
 """Dataloaders."""
 
+import abc
+from typing import Optional
 
 import torch
 
 from nemo.utils import logging
 
 
-class MegatronPretrainingSampler:
+class BaseMegatronSampler:
```

Comment:
If we get rid of the old sampler then we have to update all nemo megatron models to use dataloader_iter before merging this